src/ast/mod.rs

```diff
@@ -873,12 +873,42 @@ impl fmt::Display for Assignment {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct FunctionNamedArg {
+    pub name: Ident,
+    pub arg: Expr,
+}
+
+impl fmt::Display for FunctionNamedArg {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{} => {}", self.name, self.arg)?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum FunctionArg {
+    Named(FunctionNamedArg),
+    Unnamed(Expr),
+}
+
+impl fmt::Display for FunctionArg {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            FunctionArg::Named(named_arg) => named_arg.fmt(f),
+            FunctionArg::Unnamed(unnamed_arg) => unnamed_arg.fmt(f),
```

Comment:
We use `write!(f, "{}", ...)` in other places for this.

Reply:
fixed